js_modules/dagit/packages/core/src/partitions/PartitionStepStatus.tsx

```diff
@@ -440,10 +440,13 @@ const PartitionSquare: React.FC<{
   } else if (runs.length === 0) {
     squareStatus = 'empty';
   } else {
-    const runStatus = runs[runs.length - 1]!.status;
-    squareStatus = runStatus === RunStatus.CANCELED ? 'failure' : runStatus.toLowerCase();
+    const runStatus = [...runs].reverse().find((r) => r.status !== RunStatus.CANCELED)?.status;
```

Comment:
```
const runStatus = runs.findLast(...);
```

Reply:
https://caniuse.com/?search=findLast

Reply:
<img width="524" alt="Screenshot 2023-07-10 at 12 16 39 PM" src="https://github.com/dagster-io/dagster/assets/2286579/db57cc51-6f27-46f5-bb55-d9bdb89cf35a">
very recent feature, might be best to avoid it?

Reply:
Wow I didn't know about findLast! Super cool. I think it's probably unlikely that someone is still using a version of Chrome that is 1+ year old... but I wonder if this would be the first breaking API use? It doesn't seem like we use `findLast` anywhere else yet. 

We use `flatMap`, but that's since 2018, and `Object.fromEntries` which is since 2019 (Chrome 73). Maybe I'll hold off on this for now :/

Reply:
I know we had a user report a broken page due to using a recent `Intl` feature. I believe they were on Chrome 80ish.